src/jit/compiler.h

```diff
@@ -2366,7 +2366,8 @@ class Compiler
 
     GenTreeCall* gtNewHelperCallNode(unsigned helper, var_types type, GenTreeArgList* args = nullptr);
 
-    GenTree* gtNewLclvNode(unsigned lnum, var_types type, IL_OFFSETX ILoffs = BAD_IL_OFFSET);
+    GenTree* gtNewLclvNode(unsigned lnum, var_types type DEBUGARG(IL_OFFSETX ILoffs = BAD_IL_OFFSET));
```

Comment:
I can only assume that at some point in the past we reported lclvar reference sites to the debugger, but it doesn't appear we do currently. I wonder if that is by design, or an oversight/bug.

Reply:
This confused me for a long time. I always assumed that it has something to do with debug info (as in PDB debug info) but at same time I was wondering why do we need it on LclVar nodes when IL offsets are already stored in statement nodes and for variables there's the scope stuff.

Until I checked and found that the only use is here:
https://github.com/dotnet/coreclr/blob/6517a3519ef721d50b6b52995ca89d09b46189a2/src/jit/instr.cpp#L479-L487

under ifdef DEBUG. The emitter then stores it in `instrDescDebugInfo`, to use later in disassembly:
https://github.com/dotnet/coreclr/blob/7ba5381b95a88417034e04156da5967ab20b4e02/src/jit/emitarm64.cpp#L11531-L11536

So it looks like in this case "debug info" refers to JIT's debugging/dumping facilities and not to PDB debug info stuff.

Reply:
It's possible that someone wrote "debug info" in the comment and meant JIT internal debugging asserts/output. But I'm gonna stick with my idea that at some point in the distant past the VM or CLR/VS debugger actually wanted local var call site information. Or maybe some JIT dev thought they wanted it.

Reply:
We should chnage that to refer say "for JIT Dumps" instead of "debug info"

Reply:
Good point, updated.

And it seems that @BruceForstall is right, this might had something to do with actual debug info in the past. While looking for more comments to update, I found this comment before `siCheckVarScope`:

```
 *  In optimized code, we may not have access to gtLclVar.gtLclILoffs.
 *  So there may be ambiguity as to which entry in compiler->info.compVarScopes
 *  to use. We search the entire table and find the entry whose life
 *  begins closest to the given offset.
```
So perhaps it was used but then it was found that the offset is not maintained properly during optimizations and code using it was then removed. This code was probably a separate function, I don't think the comment actually applies to `siCheckVarScope` as that one is called with the IL offset taken from a `BasicBlock`. Which seems to make sense given what it does.